From ebf85e4cbb7271c6d61bba2ba52a1a0034abafc9 Mon Sep 17 00:00:00 2001
From: abhi128922 <abhijithjollythayyil.com>
Date: Wed, 22 Jul 2026 18:16:27 +0000
Subject: [PATCH] inputflinger: allow disabling input devices via idc

On some devices an external Daemon consumes input events and these
should not be reported to apps. Allows disabling input devices via
device.disabled property in IDC config files.

Cherry-picked from LineageOS (https://review.lineageos.org/c/436710)
---
 services/inputflinger/reader/EventHub.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/inputflinger/reader/EventHub.cpp b/services/inputflinger/reader/EventHub.cpp
index ed8d576f9e..33c84df42a 100644
--- a/services/inputflinger/reader/EventHub.cpp
+++ b/services/inputflinger/reader/EventHub.cpp
@@ -2462,6 +2462,14 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
     ALOGV("  driver:     v%d.%d.%d\n", driverVersion >> 16, (driverVersion >> 8) & 0xff,
           driverVersion & 0xff);
 
+    // Disable device if device config property set
+    if (device->configuration &&
+        device->configuration->getBool("device.disabled")) {
+        device->disable();
+        ALOGV("Disabling device with id %d
+", device->id);
+    }
+
     // Obtain the associated device, if any.
     device->associatedDevice = obtainAssociatedDeviceLocked(devicePath, device->configuration);
 
-- 
2.43.0

